src/core/execute.c

```diff
@@ -494,7 +498,7 @@ static int setup_input(
         case EXEC_INPUT_TTY:
         case EXEC_INPUT_TTY_FORCE:
         case EXEC_INPUT_TTY_FAIL: {
-                int fd;
+                int fd, r;
```

Comment:
`r` is a bit special a variable for us: we always decalre it at outermost function code, please do so here too. It's our go to variable for error return codes, and we only want it once per function, hence this rule to always declare it in outermost function scope.

Reply:
Some day I hope this advice will actually stick in my head so you don't have to repeat it all the time :(.